pkg/controller/shared_state.go

```diff
@@ -1135,20 +1135,37 @@ func (c *SharedState) cleanUpAzVolumeAttachmentByNode(ctx context.Context, azDri
 func (c *SharedState) cleanUpAzVolumeAttachments(ctx context.Context, attachments []azdiskv1beta2.AzVolumeAttachment, cleanUp attachmentCleanUpMode, caller operationRequester) error {
 	var err error
 
+	var markDetachRequest = func(attachment *azdiskv1beta2.AzVolumeAttachment, patchRequired *bool) {
+		attachment.Status.Annotations = azureutils.AddToMap(attachment.Status.Annotations, consts.VolumeDetachRequestAnnotation, string(caller))
+		*patchRequired = true
```

Comment:
`patchRequired` seems redundant here. We are unconditionally updating the annotation.

Reply:
The annotation would only be added if certain conditions are met and if annotations hasn't yet been added in prior operations. `patchRequired` is to reduce unnecessary API call. It is either we use `patchRequired` or to run a `DeepEqual` to determine if status patch is required or not.